invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/socketio/socketConnected.ts

```diff
@@ -1,33 +1,108 @@
 import { logger } from 'app/logging/logger';
-import { isInitializedChanged } from 'features/system/store/systemSlice';
+import { $baseUrl } from 'app/store/nanostores/baseUrl';
 import { size } from 'lodash-es';
+import { atom } from 'nanostores';
 import { api } from 'services/api';
+import { queueApi, selectQueueStatus } from 'services/api/endpoints/queue';
 import { receivedOpenAPISchema } from 'services/api/thunks/schema';
 import { socketConnected } from 'services/events/actions';
 
 import { startAppListening } from '../..';
 
 const log = logger('socketio');
 
+const $isFirstConnection = atom(true);
+
 export const addSocketConnectedEventListener = () => {
   startAppListening({
     actionCreator: socketConnected,
-    effect: (action, { dispatch, getState }) => {
+    effect: async (
+      action,
+      { dispatch, getState, cancelActiveListeners, delay }
+    ) => {
       log.debug('Connected');
 
-      const { nodeTemplates, config, system } = getState();
+      /**
+       * The rest of this listener has recovery logic for when the socket disconnects and reconnects.
+       *
+       * If a queue item completed during while disconnected, we need to reset the API state to re-
+       * fetch everything, updating the gallery and queue.
+       */
 
-      const { disabledTabs } = config;
+      if ($isFirstConnection.get()) {
+        // Bail on the recovery logic if this is the first connection - we don't need to recover anything
+        $isFirstConnection.set(false);
+        return;
+      }
 
-      if (!size(nodeTemplates.templates) && !disabledTabs.includes('nodes')) {
-        dispatch(receivedOpenAPISchema());
+      const { data: prevQueueStatusData } = selectQueueStatus(getState());
+
+      // Bail if queue was empty before - we have nothing to recover
```

Comment:
I'm not following this part of the logic - Isn't it still possible that generations have happened in the meantime that we need to recover?

Reply:
If the queue had nothing in progress and nothing pending when the socket disconnected, then there won't be any generations to catch up on, so the recovery logic can be totally skipped right?

It's possible that the user had a second window open somewhere and was generating there but that's an edge case that needs its own handling I think

Reply:
I've refactored this out entirely.